models/db/log.go

```diff
@@ -14,67 +14,62 @@ import (
 
 // XORMLogBridge a logger bridge from Logger to xorm
 type XORMLogBridge struct {
-	showSQLint *int32
-	logger     log.Logger
+	showSQL atomic.Bool
+	logger  log.Logger
 }
 
 // NewXORMLogger inits a log bridge for xorm
 func NewXORMLogger(showSQL bool) xormlog.Logger {
-	showSQLint := int32(0)
-	if showSQL {
-		showSQLint = 1
-	}
-	return &XORMLogBridge{
-		showSQLint: &showSQLint,
-		logger:     log.GetLogger("xorm"),
-	}
+	l := &XORMLogBridge{logger: log.GetLogger("xorm")}
+	l.showSQL.Store(showSQL)
+	return l
 }
 
 const stackLevel = 8
 
 // Log a message with defined skip and at logging level
-func (l *XORMLogBridge) Log(skip int, level log.Level, format string, v ...interface{}) error {
-	return l.logger.Log(skip+1, level, format, v...)
+func (l *XORMLogBridge) Log(skip int, level log.Level, format string, v ...interface{}) {
+	l.logger.Log(skip+1, level, format, v...)
 }
 
 // Debug show debug log
 func (l *XORMLogBridge) Debug(v ...interface{}) {
-	_ = l.Log(stackLevel, log.DEBUG, fmt.Sprint(v...))
+	l.Log(stackLevel, log.DEBUG, "%s", fmt.Sprint(v...))
```

Comment:
Is `"%s"` needed because the result of `fmt.Sprint` could produce a `%...` string?

Reply:
Yes, it is needed. `fmt.Sprint(v...)` may contain other `%xx`.

I think the old code is wrong.